lib/pluginmanager/install.rb

```diff
@@ -45,9 +45,11 @@ def validate_cli_options!
   # Check if the specified gems contains
   # the logstash `metadata`
   def verify_remote!(gems)
+    options = { :rubygems_source => gemfile.gemset.sources }
     gems.each do |plugin, version|
       puts("Validating #{[plugin, version].compact.join("-")}")
-      signal_error("Installation aborted, verification failed for #{plugin} #{version}") unless LogStash::PluginManager.logstash_plugin?(plugin, version)
+      next if LogStash::PluginManager.logstash_plugin?(plugin, version, options)
+      signal_error("Installation aborted, verification failed for #{plugin} #{version}")
     end
```

Comment:
I agree with providing better error handling, however the error you get is not from validation.

(with connection)

```
skywalker% bin/plugin install logstash-mixin-rabbitmq_connection --no-verify
Validating logstash-mixin-rabbitmq_connection
logstash-mixin-rabbitmq_connection is not a Logstash plugin
ERROR: Installation aborted, verification failed for logstash-mixin-rabbitmq_connection 
```

(without internet)

```
skywalker% bin/plugin install logstash-mixin-rabbitmq_connection --no-verify
Validating logstash-mixin-rabbitmq_connection
SocketError: recv: name or service not known
             recv_reply at /Users/purbon/work/logstash/vendor/jruby/lib/ruby/1.9/resolv.rb:782
                request at /Users/purbon/work/logstash/vendor/jruby/lib/ruby/1.9/resolv.rb:664
          each_resource at /Users/purbon/work/logstash/vendor/jruby/lib/ruby/1.9/resolv.rb:512
                 resolv at /Users/purbon/work/logstash/vendor/jruby/lib/ruby/1.9/resolv.rb:0
                   each at org/jruby/RubyArray.java:1613
```

reason is, I guess, the --no-verify appended at the end gets ignored with the mixin name you provide as example.

in the case of passing by a plugin, this is the error you get when installation fail:

```
skywalker% bin/plugin install --no-verify logstash-input-file
Installing logstash-input-file
Error Bundler::HTTPError, retrying 1/10
Could not fetch specs from https://rubygems.org/
Error Bundler::HTTPError, retrying 2/10
Could not fetch specs from https://rubygems.org/
Error Bundler::HTTPError, retrying 3/10
```

passing the right order of argument you get the same error (without internet)

```
skywalker% bin/plugin install --no-verify logstash-mixin-rabbitmq_connection
Installing logstash-mixin-rabbitmq_connection
Error Bundler::HTTPError, retrying 1/10
Could not fetch specs from https://rubygems.org/
Error Bundler::HTTPError, retrying 2/10
Could not fetch specs from https://rubygems.org/
Error Bundler::HTTPError, retrying 3/10
Could not fetch specs from https://rubygems.org/
```

investigating where to actually add the better error handling.